src/pinnable_mapped_file.cpp

```diff
@@ -216,7 +220,13 @@ pinnable_mapped_file::pinnable_mapped_file(const std::filesystem::path& dir, boo
 
 #ifndef _WIN32
       if(mode == locked) {
+#if __has_feature(address_sanitizer) && defined(__linux__)
+         //as of llvm's compilerrt 18, mlock() continues to be intercepted as a noop thus breaking tests the expect memory locking
+         // to fail. mlock2() isn't intercepted though, so use it when ASAN is enabled on Linux
+         if(mlock2(_non_file_mapped_mapping, _non_file_mapped_mapping_size, 0)) {
```

Comment:
Why not just always use `mlock2`, available since Linux 4.4, glibc 2.27.

Reply:
we'll still need an `ifdef` to accommodate macOS etc, but the `ifdef` can be moved out of line via a `MLOCK()` macro or such, so I think yeah it's probably best to do it that way